### ISSUE TITLE
Order by property value

### ISSUE THREAD

**Issue by papppeter** [OFFLINE-GmbH/oc-mall-plugin#208]
hi,

i have solution for ordering products by option value, are you interested in to include it?

**Comment by StanleyBonhomme** [comment]
Hey I'm interested. Specifically for the my theme. I want to display only the variants by "color" under the each item on the categories page.  

Here is my code:

`{# Variants Options #}


{% if item.variants.count > 0 %}
<div class="add-cart-wapper hidden-xs hidden-sm">

    <div>
        {% for value, variant in item.variants %}

        {% set variant = variant %}

        <a href="{{ productPage|page({slug: item.slug, variant: variant.hashId}) }}"
            data-mall-item-id="{{ item.prefixedId }}">

            <img src="{{ variant.main_image.thumb(50,50, 'crop') }}" alt="{{ variant.name }}" />

        </a>
        {% endfor %}

    </div>
</div>
{% endif %}`

Here is how it currently renders with the code above: 
![chrome_E3jvC6qVR1](https://user-images.githubusercontent.com/30869347/58186585-2d0e9080-7c83-11e9-9cc7-02cd7e2c7e2b.png)

Here is how I would want to render it: 
![chrome_yN4j7GCPxF](https://user-images.githubusercontent.com/30869347/58186619-3ef03380-7c83-11e9-919f-c86089e34953.png)


Basically to where it would only show "variants for the different colors" and not the "variants of the same color but different sizes"

If your solution can help me achive this that would be great!

**Comment by tobias-kuendig** [comment]
Try something like this, you'll have to make it fit your use-case though.

In your `item.htm` partial:

```php
==
<?php
function onStart()
{
    // Set the ID of the color property here
    $colorPropertyId = 4;

    // Others is a grouped collection of all available other variants.
    $this['others'] = $this['item']->variants
        ->reject(function($v) {
            // Remove the current Variant
            return $v->id === $this['item']->id;
        })
        ->groupBy(function($v) use ($colorPropertyId) {
			// Group the Variants by the property value
            $property = $v->property_values->first(function($p) use ($colorPropertyId) {
                return $p->property_id == $colorPropertyId;
            });

            if(! $property) {
                return;
            }

            return \is_array($property->value) ? json_encode($property->value) : $property->value;
        });
}
?>
==
<div>
    {% for value, variants in others %}

	    {% set variant = variants.first() %}
	
	    <a href="{{ productPage|page({slug: item.slug, variant: variant.hashId}) }}"
	        data-mall-item-id="{{ item.prefixedId }}">
	
	        <img src="{{ variant.main_image.thumb(50,50, 'crop') }}" alt="{{ variant.name }}" />
	
	    </a>

    {% endfor %}

</div>
```

The resulting structure will be something like this:

![image](https://user-images.githubusercontent.com/8600029/58230787-9c44bd00-7d35-11e9-8b09-eae4a75a338c.png)

**Comment by papppeter** [comment]
Hi,

i have a following solution:

![Screenshot_15](https://user-images.githubusercontent.com/3380880/58306378-9fa27c00-7dfb-11e9-8f6c-28005756d494.png)

https://github.com/papppeter/oc-mall-plugin/commit/249976d14825daae17ae82364752ad34556f05bc
https://github.com/papppeter/oc-mall-plugin/commit/e5331a8d026cc72dae3c61af843d7f56e5cc9758

i hope i did not forgot any of the commits.

**Comment by tobias-kuendig** [comment]
I'm closing this issue due to inactivity. If you need any further assistance feel free to reply.
